vision/cloud-client/src/main/java/com/example/vision/Detect.java

```diff
@@ -0,0 +1,405 @@
+    }
+
+    buffer.flush();
+
+    return buffer.toByteArray();
+  }
+  
+  /**
+   * Detects faces in the specified image.
+   * @param filePath The path to the file to perform face detection on.
+   * @param out A {@link PrintStream} to write detected features to.
+   * @throws IOException 
+   */
+  static public void detectFaces(String filePath, PrintStream out) throws IOException {
+    List<AnnotateImageRequest> requests = new ArrayList<>();
+
+    FileInputStream file = new FileInputStream(filePath);
+    ByteString imgBytes = ByteString.copyFrom(toByteArray(file));
+    
+    Image img = Image.newBuilder().setContent(imgBytes).build();    
+    Feature feat = Feature.newBuilder().setType(Type.FACE_DETECTION).build();
+    AnnotateImageRequest request = AnnotateImageRequest.newBuilder()
+        .addFeatures(feat)
+        .setImage(img)
+        .build();
+    requests.add(request);
+
+    BatchAnnotateImagesResponse response = visionApi.batchAnnotateImages(requests);
+    List<AnnotateImageResponse> responses = response.getResponsesList();
+
+    for (AnnotateImageResponse res : responses) {      
+      if (res.hasError()) out.printf("Error: %s\n", res.getError().getMessage());
+
+      for (FaceAnnotation annotation : res.getFaceAnnotationsList()) {
+        out.printf("anger: %s\njoy: %s\nsurprise: %s\n",
+            annotation.getAngerLikelihood(),
+            annotation.getJoyLikelihood(),
+            annotation.getSurpriseLikelihood());
+      }
+    }
+  }
+  
+  /**
+   * Detects labels in the specified image.
+   * @param filePath The path to the file to perform label detection on.
+   * @param out A {@link PrintStream} to write detected labels to.
+   * @throws IOException 
+   */
+  static public void detectLabels(String filePath, PrintStream out) throws IOException {
+    List<AnnotateImageRequest> requests = new ArrayList<>();
+
+    FileInputStream file = new FileInputStream(filePath);
+    ByteString imgBytes = ByteString.copyFrom(toByteArray(file));
+    
+    Image img = Image.newBuilder().setContent(imgBytes).build();    
+    Feature feat = Feature.newBuilder().setType(Type.LABEL_DETECTION).build();
+    AnnotateImageRequest request = AnnotateImageRequest.newBuilder()
+        .addFeatures(feat)
+        .setImage(img)
+        .build();
+    requests.add(request);
+
+    BatchAnnotateImagesResponse response = visionApi.batchAnnotateImages(requests);
+    List<AnnotateImageResponse> responses = response.getResponsesList();
+
+    for (AnnotateImageResponse res : responses) {
+      if (res.hasError()) out.printf("Error: %s\n", res.getError().getMessage());
+
+      for (EntityAnnotation annotation : res.getLabelAnnotationsList()) {
+        Map<FieldDescriptor, Object> fields = annotation.getAllFields();
+        Iterator<Entry<FieldDescriptor, Object>> iter = fields.entrySet().iterator();
+        while (iter.hasNext()) {
+            Entry<FieldDescriptor, Object> entry = iter.next();
```

Comment:
A [`foreach loop`](http://stackoverflow.com/a/46908) could save you some verbosity here.